exercises/dolica_akelloegwel/ex02_oo_basics/src/main.cpp

```diff
@@ -11,26 +11,22 @@ int main(int, char **)
 {
 	// Create various shapes
 	double squareSide = 2.5;
-	Square mySquare = Square(squareSide);
 
 	double recWidth = 2;
 	double recHeight = 5;
-	Rectangle myRectangle = Rectangle(recWidth, recHeight);
 
 	double circleRadius = 5;
-	Circle myCircle = Circle(circleRadius);
 
 	double triangleBase = 3;
 	double triangleHeight = 4;
-	Triangle myTriangle = Triangle(triangleBase, triangleHeight);
 
 	// Create a vector and place the shapes in it
-	std::vector<std::reference_wrapper<Shape> > myShapes;
+	std::vector<std::unique_ptr<Shape>> myShapes;
 
-	myShapes.push_back(mySquare);
-	myShapes.push_back(myRectangle);
-	myShapes.push_back(myCircle);
-	myShapes.push_back(myTriangle);
+	myShapes.push_back(std::make_unique<Square>(squareSide));

```

Comment:
I would either change the declarations above to be `const`, i.e. `circleRadius`, or better yet, change the push_backs to include in-line comments:
```cpp
myShapes.push_back(std::make_unique<Rectangle>(2 /*recWidth*/, 5 /*recHeight*/))```